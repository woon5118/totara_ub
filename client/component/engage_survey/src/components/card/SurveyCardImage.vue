<!--
  This file is part of Totara Enterprise Extensions.

  Copyright (C) 2020 onwards Totara Learning Solutions LTD

  Totara Enterprise Extensions is provided only to Totara
  Learning Solutions LTD's customers and partners, pursuant to
  the terms and conditions of a separate agreement with Totara
  Learning Solutions LTD or its affiliate.

  If you do not have an agreement with Totara Learning Solutions
  LTD, you may not access, use, modify, or distribute this software.
  Please contact [licensing@totaralearning.com] for more information.

  @author Cody Finegan <cody.finegan@totaralearning.com>
  @module engage_survey
-->

<template>
  <div class="tui-engageSurveyImage">
    <SurveyIcon
      :image-alt="$str('image_alt', 'engage_survey', name)"
      class="tui-engageSurveyImage__img"
    />
  </div>
</template>

<script>
import { imageMixin } from 'totara_engage/index';
import SurveyIcon from 'engage_survey/components/icon/SurveyIcon';

export default {
  components: {
    SurveyIcon,
  },

  mixins: [imageMixin],
};
</script>

<lang-strings>
  {
    "engage_survey": [
      "image_alt"
    ]
  }
</lang-strings>

<style lang="scss">
.tui-engageSurveyImage {
  &__img {
    width: 100%;
    height: 100%;
  }
}
</style>
