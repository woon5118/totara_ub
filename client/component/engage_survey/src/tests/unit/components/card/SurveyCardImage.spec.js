/**
 * This file is part of Totara Enterprise Extensions.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise Extensions is provided only to Totara
 * Learning Solutions LTD's customers and partners, pursuant to
 * the terms and conditions of a separate agreement with Totara
 * Learning Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [licensing@totaralearning.com] for more information.
 *
 * @author Cody Finegan <cody.finegan@totaralearning.com>
 * @module engage_survey
 */

import { shallowMount } from '@vue/test-utils';
import SurveyCardImage from 'engage_survey/components/card/SurveyCardImage';

describe('engage_survey/components/card/SurveyCardImage.vue', () => {
  let wrapper = null;

  beforeAll(() => {
    wrapper = shallowMount(SurveyCardImage, {
      mocks: {
        $str(identifier, component, param) {
          return `[${identifier}, ${component} - ${param}]`;
        },
        $url(str) {
          return str;
        },
      },

      propsData: {
        instanceId: 1,
        name: 'Hello world ',
        image: null,
      },
    });
  });

  it('Checks snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
