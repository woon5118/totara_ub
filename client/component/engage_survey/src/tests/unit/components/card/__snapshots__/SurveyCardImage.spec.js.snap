// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`engage_survey/components/card/SurveyCardImage.vue Checks snapshot 1`] = `
<div
  class="tui-engageSurveyImage"
>
  <surveyicon-stub
    class="tui-engageSurveyImage__img"
    imagealt="[image_alt, engage_survey - Hello world ]"
  />
</div>
`;
